lib/util.js

```diff
@@ -59,32 +63,44 @@ exports.getLabels = function (labelNames, args) {
 	return acc;
 };
 
-function hashObject(labels) {
+function fastHashObject(keys, labels) {
+	if (keys.length === 0) {
+		return '';
+	}
+
+	let hash = '';
+
+	for (let i = 0; i < keys.length; i++) {
+		const key = keys[i];
+		const value = labels[key];
+		if (typeof value === 'undefined') continue;
+
+		hash += `${key}:${value},`;
+	}
+
+	return hash;
+}
+
+function hashObject(labels, labelNames) {
 	// We don't actually need a hash here. We just need a string that
 	// is unique for each possible labels object and consistent across
 	// calls with equivalent labels objects.
-	let keys = Object.keys(labels);
-	if (keys.length === 0) {
-		return '';
+
+	if (labelNames) {
+		return fastHashObject(labelNames, labels);
 	}
-	// else
+
+	let keys = Object.keys(labels);
 	if (keys.length > 1) {
 		keys = keys.sort(); // need consistency across calls
```

Comment:
If you're making other changes, you could remove the useless `keys = ` bit for clarity.

Reply:
Are you suggesting to change the code to:
```javascript
let keys = Object.keys(labels);
if (keys.length > 1) {
	return fastHashObject(keys.sort(), labels)
} else {
	return fastHashObject(keys, labels))
}
```

Reply:
removed the assignment.